components/filedata-table.vue

```diff
@@ -29,21 +62,37 @@
 				let tableRow = document.getElementById(itemID);
 
 				// 1. Determine if this item is to be colored or uncolored
-				if ( this.defaultPalette.bColor == tableRow.style.backgroundColor ||
-					 "" == tableRow.style.backgroundColor ) {
-					tableRow.style.backgroundColor = this.currentPalette.bColor;
-					tableRow.style.color = this.currentPalette.fColor;
+				if ( tableRow.classList.contains(this.defaultStyleClass) ) {
+
+					// A. Replace the default style class name with the desired new style
+					tableRow.classList.replace(this.defaultStyleClass, this.currentStyleClass);
 				} else {
-					tableRow.style.backgroundColor = this.defaultPalette.bColor;
-					tableRow.style.color = this.defaultPalette.fColor;
+
+					// A. If a category style class exists, replace it with the desired new style
+					let oldStyleClass = "";
+					for ( let index = 0; index < tableRow.classList.length; index++ ) {
+
+						if ( -1 != tableRow.classList[index].indexOf(this.$store.getters.cssStylePrefix) ) {
+
+							oldStyleClass = tableRow.classList[index];
+							break;
+						}
+					}
+
+					// B. Determine if category link is being replaced or column is being unlinked
+					if ( oldStyleClass == this.currentStyleClass ) {
+						tableRow.classList.replace(oldStyleClass, this.defaultStyleClass);
+					} else {
+						tableRow.classList.replace(oldStyleClass, this.currentStyleClass);
+					}
 				}
 
-				// 2. Send the column name back to the parent to link it with the current category
+				// 3. Send the column name back to the parent to link it with the current category
 				this.$emit("column-name-selected", this.tableData[p_index]);
 			}
 		},
 
-		props: ["currentPalette", "defaultPalette", "fields", "tableData", "tableID"]
+		props: ["currentStyleClass", "defaultStyleClass", "fields", "tableData", "tableID"]
```

Comment:
I understand that we need to pass the selected category between the category-selector-component and here, but I would suggest that we pass the category name rather than the stylesheet class that belongs to this category. Then we can just compare the actual column-category assignment and don't have to infer it from the current styling of the rows.